src/exclusion_filter.cpp

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include <exclusion_filter.hpp>
+
+namespace ddwaf {
+
+bool exclusion_filter::match(const object_store& store,
+    const ddwaf::manifest &manifest,
+    cache_type &cache, ddwaf::timer& deadline) const
+{
+    if (cache.result) { return true; }
+
+    for (auto cond : conditions_) {
+        // If there's a (false) cache hit, we only need to run this condition
+        // on new parameters.
```

Comment:
If there is at most one false condition whose result was previously recorded (because you return as soon as something doesn't match), you could store instead just the first unmatched conditions. All the previous ones were matched, the stored one we tried and it didn't match (so we need only run on new parameters) and the remaining ones need to be run with all the parameters.

Reply:
I will update this in the subsequent PR if that's okay.